components/admin_console/team_channel_settings/channel/list/channel_list.tsx

```diff
@@ -2,182 +2,227 @@
+            loading: false,
+            term: '',
             channels: [],
-            searchTotalCount: 0,
-            pageResetKey: 0,
-            searchMode: false,
+            page: 0,
+            total: 0,
         };
     }
 
-    searchBar = () => {
-        return (
-            <div className='groups-list--global-actions'>
-                <div className='group-list-search'>
-                    <input
-                        type='text'
-                        placeholder={Utils.localizeMessage('search_bar.search', 'Search')}
-                        onKeyUp={this.handleChannelSearchKeyUp}
-                        onChange={this.searchBarChangeHandler}
-                        value={this.state.searchString}
-                        data-testid='search-input'
-                    />
-                    <SearchIcon
-                        id='searchIcon'
-                        className='search__icon'
-                        aria-hidden='true'
-                    />
-                    <i
-                        className={'fa fa-times-circle group-filter-action ' + (this.state.searchString.length ? '' : 'hidden')}
-                        onClick={this.resetSearch}
-                        data-testid='clear-search'
-                    />
-                </div>
-            </div>
-        );
-    };
+    componentDidMount() {
+        this.loadPage();
+    }
 
-    private searchBarChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
-        const {searchString} = this.state;
-        if (searchString.length !== 0 && e.target.value.length === 0) {
-            this.resetSearch();
-            return;
-        }
-        this.setState({searchString: e.target.value});
-    };
+    getPaginationProps = () => {
+        const {page, term} = this.state;
+        const total = term === '' ? this.props.total : this.state.total;
+        const startCount = (page * PAGE_SIZE) + 1;
+        let endCount = (page + 1) * PAGE_SIZE;
+        endCount = endCount > total ? total : endCount;
+        return {startCount, endCount, total};
+    }
+
+    loadPage = async (page = 0, term = '') => {
+        this.setState({loading: true, term});
 
-    private handleChannelSearchKeyUp = async (e: React.KeyboardEvent) => {
-        const {key} = e;
-        const {searchString} = this.state;
-        if (key === Constants.KeyCodes.ENTER[0]) {
-            if (searchString.length > 1) {
-                const response = await this.props.actions.searchAllChannels(searchString, '', false, 0, PAGE_SIZE);
-                if ('data' in response) {
-                    this.setState({searchMode: true, channels: response.data.channels, searchTotalCount: response.data.total_count, pageResetKey: Date.now()});
-                }
+        if (term.length > 0) {
+            if (page > 0) {
+                this.searchChannels(page, term);
+            } else {
+                this.searchChannelsDebounced(page, term);
```

Comment:
what happens if someone is already on page 3 and then they start searching wouldnt this result in a non debounced search? - does the page get reset to 0 when the search term changes?

Reply:
Looks like it does